components/tooltip/Tooltip.js

```diff
@@ -0,0 +1,271 @@
+          this.setState({ active: false });
+        } else if (this.state.visible) {
+          this.setState({ visible: false });
+        }
+      }
+
+      calculatePosition(element) {
+        const { top, left, height, width } = element.getBoundingClientRect();
+        const position = this.getPosition(element);
+        const xOffset = window.scrollX || window.pageXOffset;
+        const yOffset = window.scrollY || window.pageYOffset;
+
+        if (position === POSITION.BOTTOM) {
+          return {
+            top: top + height + yOffset,
+            left: left + width / 2 + xOffset,
+            position,
+          };
+        } else if (position === POSITION.TOP) {
+          return {
+            top: top + yOffset,
+            left: left + width / 2 + xOffset,
+            position,
+          };
+        } else if (position === POSITION.LEFT) {
+          return {
+            top: top + height / 2 + yOffset,
+            left: left + xOffset,
+            position,
+          };
+        } else if (position === POSITION.RIGHT) {
+          return {
+            top: top + height / 2 + yOffset,
+            left: left + width + xOffset,
+            position,
+          };
+        }
+
+        return undefined;
+      }
+
+      handleMouseEnter = event => {
+        this.activate(this.calculatePosition(event.currentTarget));
+
+        if (this.props.onMouseEnter) {
+          this.props.onMouseEnter(event);
+        }
+      };
+
+      handleMouseLeave = event => {
+        this.deactivate();
+
+        if (this.props.onMouseLeave) {
+          this.props.onMouseLeave(event);
+        }
+      };
+
+      handleClick = event => {
+        if (this.props.tooltipHideOnClick && this.state.active) {
+          this.deactivate();
+        }
+
+        if (this.props.tooltipShowOnClick && !this.state.active) {
+          this.activate(this.calculatePosition(event.currentTarget));
+        }
+
+        if (this.props.onClick) {
+          this.props.onClick(event);
+        }
+      };
+
+      render() {
+        const { active, left, top, position, visible } = this.state;
+        const positionClass = `tooltip${position.charAt(0).toUpperCase() + position.slice(1)}`;
+        const {
+          children,
+          className,
+          onClick, // eslint-disable-line no-unused-vars
+          onMouseEnter, // eslint-disable-line no-unused-vars
+          onMouseLeave, // eslint-disable-line no-unused-vars
+          tooltip,
+          tooltipColor,
+          tooltipDelay, // eslint-disable-line no-unused-vars
+          tooltipHideOnClick, // eslint-disable-line no-unused-vars
+          tooltipIcon,
+          tooltipPosition, // eslint-disable-line no-unused-vars
+          tooltipShowOnClick, // eslint-disable-line no-unused-vars
+          tooltipSize,
+          ...other
+        } = this.props;
+
+        const _className = cx(theme.tooltip, theme[tooltipColor], theme[tooltipSize], {
```

Comment:
We're never using underscores, I think you mean 
```js
const classNames = cx(...);
```
to be consistent with the rest of the codebase?